src/data_info/services/stat.clj

```diff
@@ -46,76 +65,107 @@
     stat-map))
 
-
 (defn ^IPersistentMap decorate-stat
-  [^IPersistentMap cm ^String user ^IPersistentMap stat]
+  [^IPersistentMap cm ^String user ^IPersistentMap stat included-keys]
   (let [path (:path stat)]
     (-> stat
-      (assoc :id         (-> (meta/get-attribute cm path uuid/uuid-attr) first :value)
-             :permission (perm/permission-for cm user path))
-      (merge-label user path)
-      (merge-type-info cm user path)
-      (merge-shares cm user path)
-      (merge-counts cm user path))))
-
+      (assoc :id         (if (needs-key? included-keys :id) (-> (meta/get-attribute cm path uuid/uuid-attr) first :value))
+             :permission (if (needs-key? included-keys :permission) (perm/permission-for cm user path)))
+      (merge-label user path included-keys)
+      (merge-type-info cm user path included-keys)
+      (merge-shares cm user path included-keys)
+      (merge-counts cm user path included-keys)
+      (select-keys included-keys))))
+
+(defn- get-filter-set
+  [filter-vec-or-string default]
+  (if (nil? filter-vec-or-string)
+    (set default)
+    (if (string? filter-vec-or-string)
+      (set (map keyword (string/split filter-vec-or-string #",")))
+      (set (map keyword filter-vec-or-string)))))
+
+(defn- process-filters
+  "Process an include and an exclude string into just a list of keys to include"
+  [include exclude]
+  (let [all-keys #{:id :path :type :label :date-created :date-modified :permission :share-count :file-count :dir-count :file-size :content-type :infoType :md5}
+        includes-set (get-filter-set include all-keys)
+        excludes-set (get-filter-set exclude [])]
+      (cset/intersection all-keys (cset/difference includes-set excludes-set))))
 
 (defn ^IPersistentMap path-stat
-  [^IPersistentMap cm ^String user ^String path]
-  (let [path (ft/rm-last-slash path)]
+  [^IPersistentMap cm ^String user ^String path & {:keys [filter-include filter-exclude] :or {filter-include nil filter-exclude nil}}]
+  (let [path (ft/rm-last-slash path)
+        included-keys (process-filters filter-include filter-exclude)]
     (log/debug "[path-stat] user:" user "path:" path)
     (validators/path-exists cm path)
-    (decorate-stat cm user (info/stat cm path))))
+    (let [base-stat (if (needs-any-key? included-keys :type :date-created :date-modified :file-size :md5)
+                      (info/stat cm path)
+                      {:path path})]
+      (decorate-stat cm user base-stat included-keys))))
 
 (defn ^IPersistentMap uuid-stat
-  [^IPersistentMap cm ^String user uuid]
+  [^IPersistentMap cm ^String user uuid & {:keys [filter-include filter-exclude] :or {filter-include nil filter-exclude nil}}]
   (log/debug "[uuid-stat] user:" user "uuid:" uuid)
   (let [path (uuid/get-path cm uuid)]
-    (path-stat cm user path)))
+    (path-stat cm user path :filter-include filter-include :filter-exclude filter-exclude)))
```

Comment:
One optimization I haven't done because uuid-stat is fairly unused is making it so that if we come in via this route, we don't need to do an extra fetch for the ID (which was, of course, provided). I may look into that before merging, though.